core/modules/rproc/InfileMerger.cc

```diff
@@ -280,11 +169,17 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
             return false;
         }
     }
-    return _importResponse(response);
+
+    // Nothing to do if size is zero.
+    if (response->result.row_size() != 0) {
```

Comment:
:nitpick: (damn, no emoticon for nitpick) I'd probably move this check into `_merge()` to make it more generic.